src/components/Login/Login.jsx

```diff
@@ -0,0 +1,252 @@
+import React, { useEffect, useState } from "react";
+
+import "./Login.css";
+import wavingPikachu from "../../img/waving-pikachu.png";
+import thorEndgame from "../../img/thor-endgame.png";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faUser } from "@fortawesome/free-solid-svg-icons";
+import { faLock } from "@fortawesome/free-solid-svg-icons";
+import { faFacebookF } from "@fortawesome/free-brands-svg-icons";
+import { faGoogle } from "@fortawesome/free-brands-svg-icons";
+import { faEnvelope } from "@fortawesome/free-solid-svg-icons";
+import { auth, gProvider, fProvider } from "../../fire";
+import { useDataLayerValue } from "../../context/DataLayer";
+
+function Login() {
+  const [{ user }, dispatch] = useDataLayerValue();
+  const [elementsObj, setElementsObj] = useState({});
+  const [userObj, setUserObj] = useState({});
+
+  const addSignUpMode = () => {
+    elementsObj.container.classList.add("sign-up-mode");
+  };
+
+  const remSignUpMode = () => {
+    elementsObj.container.classList.remove("sign-up-mode");
+  };
+
+  const handleSignUp = (e) => {
+    e.preventDefault();
+    auth
+      .createUserWithEmailAndPassword(userObj.email, userObj.password)
+      .then((user) => {
+        // signed up
+        var u = auth.currentUser;
```

Comment:
lets use a better naming convention than u.  you could just destructure currentUser from auth... 

`const { currentUser } = auth;`